daemon/src/peer/run_state.rs

```diff
@@ -199,7 +199,7 @@ impl RunState {
     /// new state and in some cases produes commands which should be
     /// executed in the appropriate subroutines.
     pub fn transition(&mut self, input: Input) -> Vec<Command> {
```

Comment:
Might be worth add ing a `#[instrument]` to this function and removing the "start" and "end" traces as they serve the same purpose.

Reply:
We’re also printing `self.status` and the resulting `cmds` which would not be possible with `instrument`. Is that ok?

Reply:
I think you should be able to do that with something like (based on my reading of https://docs.rs/tracing/0.1.26/tracing/attr.instrument.html#examples-1):

```rust
#[instrument(skip(self), fields(status=self.status)]
```

Reply:
Nice! https://github.com/radicle-dev/radicle-link/pull/727/commits/e29d2bd7c682e38f55a5c68d1cdfa9ef28525609

Reply:
Ugh, I'm actually wrong about this. The `instrument` macro generates a span which records the fields when the span is entered. I forgot that the `status` changes as a result of the `transition` and that won't be recorded by the span. Your original code was better I think. My apologies.

Reply:
Reverted in ea907e4862c809649133b12ef0fa01299fd43d40